File: fifa2014group.py
#Daniel Levine, Flanders Lorton

# Given the letter of a group, uses the API to find the standings of that group
import sys,requests,json

   
try: 
   results = requests.get("http://worldcup.kimonolabs.com/api/teams?apikey=H5KHBD8rdv859QBLzGNswxxKOUIDdg5o&group="+sys.argv[1]+"&fields=name,groupRank,wins,losses,draws")
   results= results.json()
   results=sorted(results, key = lambda group: (group['groupRank']))
   print "Team".ljust(17),"W".ljust(2),"D".ljust(2),"L".ljust(2)
   for team in results:
   	   print team['name'].ljust(17),repr(team['wins']).ljust(2),repr(team['draws']).ljust(2),repr(team['losses']).ljust(2)
except:
	print "Need just one commandline argument, A...H"

File: homework2/fifa2014group.js
//takes use argument then uses the kimono api to get a group that the user chose
//then returns the group in order of group ranking


var request = require("request");


var apikey="H5KHBD8rdv859QBLzGNswxxKOUIDdg5o"

if((process.argv[2].length>=2)||(/[^A-H]/).test(process.argv[2])){
	console.log("Need just one commandline argument, A...H")
	process.exit()
}


request("http://worldcup.kimonolabs.com/api/teams?apikey="+apikey+"&group="+process.argv[2]+"&fields=name,groupRank,wins,losses,draws&sort=groupRank", 
  function(err,response, body) {
  if (!err && response.statusCode == 200) {
	  	var group = JSON.parse(body);
	  	console.log("Name                    W | D | L")
	  	for(var i = 0; i < group.length; i++) {

		console.log(group[i]["name"]+"            "+group[i]["wins"]+"   "+group[i]["draws"]+"   "+group[i]["losses"]);
	}
 }   
});
